backend/backup_media.js

```diff
@@ -0,0 +1,151 @@
+require('dotenv-flow').config();
+const mysql = require('mysql2');
+const fs = require('fs');
+const crypto = require('crypto');
+const http = require('http');
+const https = require('https');
+const sizeOf = require('image-size');
+
+const MEDIA_TYPE_IMG = 'img';
+const MEDIA_TYPE_VIDEO = 'video';
+
+const config = {
+    host: process.env.MYSQL_HOST || "localhost",
+    port: parseInt(process.env.MYSQL_PORT) || 3306,
+    user: "root",
+    password: process.env.MYSQL_ROOT_PASSWORD || "orbitar",
+    database: process.env.MYSQL_DATABASE || "orbitar_db",
+    dir: './media-backup',
+    dbFile: './media-backup/index.csv'
+};
+
+const sleep = function(ms) {
+  return new Promise(function(resolve) {
+    setTimeout(resolve, ms);
+  });
+}
+
+const download = function(url, dest, cb) {
+  if (url.match(/^\/\//)) {
+    url = 'https:' + url;
+  }
+  try {
+    const file = fs.createWriteStream(dest);
+    let loader = http;
+    if (url.includes('https://')) {
+      loader = https;
+    }
+    loader.get(url, function(response) {
+      response.pipe(file);
+      file.on('finish', function() {
+        file.close(cb);
+      });
+    }).on('error', function(err) {
+      console.trace(err);
+      fs.unlinkSync(dest);
+      if (cb) cb(err.message);
+    });
+  } catch (e) {
+    console.trace(e);
+  }
+};
+
+try { fs.mkdirSync(config.dir); } catch (e) {}
+if (!fs.existsSync(config.dbFile)) {
+  fs.openSync(config.dbFile, 'w+');
+  fs.appendFileSync(config.dbFile, 'item_type,item_id,media_type,media_src,media_id,img_width,img_height  ' + "\n");
+} else {
+  fs.openSync(config.dbFile, 'w+');
+}
+
+const connection = mysql.createConnection({
+  host: config.host,
+  user: config.user,
+  password: config.password,
+  database: config.database
+});
+
+const entries = [];
+
+connection.query(
```

Comment:
Fetching everything each run is very expensive (especially using text-base queries).

To run this in production we need to store the update timestamp of the latest processed entity and start the next batch from there. Also fetch and process entities in chunks.